iceoryx_binding_c/test/moduletests/test_client.cpp

```diff
@@ -0,0 +1,492 @@
+    EXPECT_THAT(iox_client_loan_aligned_request(sut, &payload, 32, 32), Eq(AllocationResult_SUCCESS));
+
+    payload = nullptr;
+    EXPECT_THAT(iox_client_loan_request(sut, &payload, 322), Eq(AllocationResult_RUNNING_OUT_OF_CHUNKS));
+    EXPECT_THAT(payload, Eq(nullptr));
+    EXPECT_THAT(memoryManager.getMemPoolInfo(0).m_usedChunks, Eq(2U));
+}
+
+TEST_F(iox_client_test, ReleaseWorksOnValidPayload)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "159e9b42-4f43-41df-8449-5891950eb592");
+    prepareClientInit();
+    iox_client_t sut = iox_client_init(&sutStorage, SERVICE, INSTANCE, EVENT, nullptr);
+
+    void* payload = nullptr;
+    iox_client_loan_request(sut, &payload, 32);
+
+    iox_client_release_request(sut, payload);
+
+    EXPECT_THAT(memoryManager.getMemPoolInfo(0).m_usedChunks, Eq(0U));
+}
+
+TEST_F(iox_client_test, LoanAndSendWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "fd83a4cd-6f86-47f8-aa46-e20f34959461");
+    prepareClientInit();
+    iox_client_t sut = iox_client_init(&sutStorage, SERVICE, INSTANCE, EVENT, nullptr);
+    connect();
+
+    void* payload = nullptr;
+    EXPECT_THAT(iox_client_loan_request(sut, &payload, sizeof(int64_t)), Eq(AllocationResult_SUCCESS));
+    *static_cast<int64_t*>(payload) = 8912389;
+
+    iox_client_send(sut, payload);
+
+    serverRequestQueue.tryPop()
+        .and_then([&](auto& sharedChunk) {
+            auto msg = static_cast<int64_t*>(sharedChunk.getUserPayload());
+            EXPECT_THAT(*msg, Eq(8912389));
+        })
+        .or_else([&] { GTEST_FAIL() << "Expected request but got none"; });
+}
+
+TEST_F(iox_client_test, ConnectWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "4809d1ce-3a53-4981-bded-4709599f62b5");
+    prepareClientInit();
+    iox_client_t sut = iox_client_init(&sutStorage, SERVICE, INSTANCE, EVENT, nullptr);
+    iox_client_connect(sut);
+
+    EXPECT_THAT(sutPort->m_connectRequested.load(), Eq(true));
+}
+
+TEST_F(iox_client_test, DisconnectWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "ebb07dc5-1aa2-4dbe-8b86-7378e6bd2ed2");
+    prepareClientInit();
+    iox_client_t sut = iox_client_init(&sutStorage, SERVICE, INSTANCE, EVENT, nullptr);
+    iox_client_connect(sut);
+    iox_client_disconnect(sut);
+
+    EXPECT_THAT(sutPort->m_connectRequested.load(), Eq(false));
+}
+
+TEST_F(iox_client_test, GetConnectionIsNotConnectedWhenCreatedWithoutAutoConnect)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "8689930e-5ce8-4d1e-8863-01689d6d0fbd");
+    iox_client_options_t options;
+    iox_client_options_init(&options);
+    options.connectOnCreate = false;
+
+    ClientOptions cppOptions;
+    cppOptions.connectOnCreate = false;
+    prepareClientInit(cppOptions);
+    iox_client_t sut = iox_client_init(&sutStorage, SERVICE, INSTANCE, EVENT, &options);
+
+    EXPECT_THAT(iox_client_get_connection_state(sut), Eq(ConnectionState_NOT_CONNECTED));
+}
+
+TEST_F(iox_client_test, GetConnectionReturnsConnectRequested)
```

Comment:
You can combine this and the following three test cases. They do not add value since the same code paths are called all the time and you already tested the enum translation.